docs/proposals/user-impersonation-proposal.md

```diff
@@ -0,0 +1,39 @@
+# User Impersonation Proposal
+
+## Introduction
+We will need to add some security settings to help administrators lock down the broker. For this, we are going to check for privilege escalation and will move the location of where the APB pod will run.
+
+## Problem Description
+The problem is that currently privilege escalation is a concern for users who have access to the broker. We want to give the cluster admin who is setting up the service broker to have assurances that the broker can have some safety assurances.  
+
+## Implementation Details
+We will need to do 5 things to satisfy the requirements.
+1. Check if the user has the permissions to cover the cluster role to be used.
+2. If the cluster admin while setting up the broker has set a config value to auto escalate we will immediately continue and not check the user's permissions.
+3. Create transient namespace/project
```

Comment:
Instead of a transient namespace, should we create an APB only namespace that only can be accessed by the broker? Do we gain anything by deleting and creating a namespace per APB?

Reply:
I think we can an extra layer of isolation. In practice, I don't know if that matters. I could see this being a configuration value down the road if you don't want the overhead.

Reply:
We were advised to proceed with the transient namespace per aos-security, ensures cleaner workflow.  

Making this a config option in future sound good, that is not a requirement for 3.7 timeframe.